server/config.go

```diff
@@ -0,0 +1,85 @@
+package server
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path"
+)
+
+const defaultGatewayConfigPath = "~/.dfms/gateway_cfg.json"
+
+func init() {
+	p := resolvePath(defaultGatewayConfigPath)
+	_, err := os.Stat(p)
+	if os.IsNotExist(err) {
+		err = saveConfig(DefaultConfig(), p)
+		if err != nil {
+			log.Fatalf("Save default config: ", err)
+		}
+	} else if err != nil {
+		log.Fatal(err)
+	}
+}
+
+type Config struct {
+	Name        string
+	Address     string
+	ApiAddress  string
+	GetOnly     bool
+	LogAllError bool
+}
+
+func DefaultConfig() *Config {
+	return &Config{
+		Name:        "DFMS Gateway",
+		Address:     ":5000",
+		ApiAddress:  "http://localhost:6366",
+		GetOnly:     true,
+		LogAllError: false,
+	}
+}
```

Comment:
If there is no usage outside the package for Config struct and default constructor, there is no need to make it public.